### ISSUE TITLE
Example showing how to export to CSV

### ISSUE THREAD

**Issue by wihl** [googleads/google-ads-python#235]
In the AdWords API, report export to CSV was possible. By design, the Google Ads API returns [row objects](https://developers.google.com/google-ads/api/reference/rpc/google.ads.googleads.v3.services#google.ads.googleads.v3.services.GoogleAdsRow). Since CSV export is a common use case, illustrate an example of how to do this in Python.

([Ref.](https://groups.google.com/forum/#!msg/adwords-api/zS9QwFe-HGE/pOhMD9psCgAJ))

**Comment by davedavis** [comment]
@wihl This is a bit old, and I'd love to take a stab at it, however, I'm a little concerned about how appropriate this is in the library/repo? 

If you still think this is a common enough use case to warrant an example, are you happy for me to go ahead with using a basic query along with search_stream and the PEP 305 CSV File API along with a standard dialect?

**Comment by wihl** [comment]
Sure I think it would still be valuable. When ready, submit as a Pull Request. Here's a messy example which could be significantly improved:

```
#!/usr/bin/env python

import csv
import sys

from google.ads.google_ads.client import GoogleAdsClient
from google.ads.google_ads.errors import GoogleAdsException


def main(client, customer_id, strategy_id, write_headers):
    ga_service = client.get_service("GoogleAdsService", version="v6")
    query = f"""
        SELECT
            bidding_strategy.id,
            segments.date,
            segments.hour,
            metrics.cost_micros,
            metrics.impressions,
            bidding_strategy.target_impression_share.location,
            metrics.clicks,
            metrics.conversions,
            metrics.conversions_value
        FROM
            bidding_strategy
        WHERE
            segments.date >= '2020-11-02'
        AND segments.date <= '2020-11-08'
        AND bidding_strategy.id = {str(strategy_id)}
        ORDER BY segments.date, segments.hour
        """

    response = ga_service.search_stream(customer_id, query=query)

    try:
        count = 0
        csvfile = open("output.csv", "w", newline="")
        csvw = csv.writer(csvfile, delimiter=",")
        if write_headers:
            csvw.writerow(
                [
                    "Portfolio ID",
                    "Date",
                    "Hour",
                    "Cost(micros)",
                    "Impression",
                    "Target Impression Location",
                    "Clicks",
                    "Conversions",
                    "Conversion Value",
                ]
            )
        for batch in response:
          for row in batch.results:
            count += 1
            csvw.writerow(
                [
                    row.bidding_strategy.id,
                    row.segments.date,
                    row.segments.hour,
                    row.metrics.cost_micros,
                    row.metrics.impressions,
                    row.bidding_strategy.target_impression_share.location,
                    row.metrics.clicks,
                    row.metrics.conversions,
                    row.metrics.conversions_value,
                ]
            )
        print(
            f"Customer {customer_id}, Portfolio {strategy_id}, "
            f"Total count: {count}"
        )
    except GoogleAdsException as ex:
        print(
            f'Request with ID "{ex.request_id}" failed with status '
            f'"{ex.error.code().name}" and includes the following errors:'
        )
        for error in ex.failure.errors:
            print(f'\tError with message "{error.message}".')
            if error.location:
                for field_path_element in error.location.field_path_elements:
                    print(f"\t\tOn field: {field_path_element.field_name}")
        sys.exit(1)


if __name__ == "__main__":
    # GoogleAdsClient will read the google-ads.yaml configuration file in the
    # home directory if none is specified.
    google_ads_client = GoogleAdsClient.load_from_storage()

    main(google_ads_client, <CUSTOMER ID>, <STRATEGY ID>, True)
```

**Comment by BenRKarl** [comment]
@wihl should the new example get filed under `examples/misc` or `examples/reporting`?

**Comment by wihl** [comment]
`examples\reporting` as this would be more consistent with the AdWords API examples that export to CSV which are primarily (but not exclusively) in `reporting`

**Comment by davedavis** [comment]
Perfect. Thanks for the guidance! I'll get working on this in the next week.

**Comment by davedavis** [comment]
Hi @wihl and @BenRKarl Before submitting the PR, I was wondering if you could help with a couple of questions that I'd rather keep out of any review. 

Firstly, perhaps I'm overthinking this, but I wanted to add a dynamic way to add the header row to the CSV rather than hard code it like it the example above. It seems the only way to do this is to grab the field_masks.paths from the response. Is this an appropriate way to use the field masks, as I know this is not what they're for?

Secondly, adding the field masks and summary row (which is returned last) means that I have to store all the rows in a separate list until the summary row is received. Then write the headers, data rows and then the summary. Is this over-complicating things or should I just clean up the example above and let the user hard code the values they want? 

Thanks a million.

**Comment by wihl** [comment]
Given the GAQL is hard-coded, I don't think you need a fancy way to retrieve the column names. If someone is dynamically generating GAQL, it would be additionally trivial to generate the column names. One additional factor is that only English comes back from the API call, but users may want to specify column names in a different (human) language.

Wrt to the summary row, we are making changes to the API to make the summary row readily available. This will be available in the coming months, so don't make any special considerations for it now, when it will be simplified soon.


In summary, no need to add these additional complexities. Thanks for your efforts!

**Comment by davedavis** [comment]
Thanks for your patience @BenRKarl 
I've updated the PR. I'm not sure I did the review correctly. But all the changes requested or change in comments are submitted in the latest commit.
